src/main/scala/chiseltest/legacy/backends/verilator/VerilatorCompilerPhase.scala

```diff
@@ -0,0 +1,258 @@
+package chiseltest.stage
+
+import java.io.{File, FileWriter, IOException}
+import java.nio.file.StandardCopyOption.REPLACE_EXISTING
+import java.nio.file.{FileAlreadyExistsException, Files, Paths}
+
+import chiseltest.internal.WriteVcdAnnotation
+import chiseltest.legacy.backends.verilator.{CommandEditsFile, TestCommandOverride, VerilatorCFlags, VerilatorFlags, verilogToVerilator}
+import firrtl.AnnotationSeq
+import firrtl.annotations.NoTargetAnnotation
+import firrtl.ir._
+import firrtl.options.{Phase, TargetDirAnnotation}
+import firrtl.stage.FirrtlCircuitAnnotation
+
+import scala.sys.process.{ProcessBuilder, ProcessLogger, _}
+
+case class CommandAnnotation(value: Seq[String]) extends NoTargetAnnotation
+
+class VerilatorCompilerPhase extends Phase {
```

Comment:
I think moving to stage/phase approach is good but this could use a general outline of what this phase does.